### PR TITLE
release: v2

### PR DESCRIPTION
**Pull request by chloelbn**
This PR updates the dependency to the `algoliasearch` gem to `algolia` gem. 
This will be a new major considering the breaking changes introduced in `algolia` new gem.

TODO: update Algolia's doc

### CODE REVIEW

File: algoliasearch-rails.gemspec [hunk 1]
```diff
@@ -79,19 +79,18 @@ Gem::Specification.new do |s|
 
     if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('1.2.0') then
       s.add_runtime_dependency(%q<json>, [">= 1.5.1"])
-      s.add_runtime_dependency(%q<algoliasearch>, [">= 1.26.0", "< 2.0.0"])
+      s.add_runtime_dependency(%q<algolia>, ["< 3.0.0"])
       s.add_development_dependency(%q<will_paginate>, [">= 2.3.15"])
-      s.add_development_dependency(%q<kaminari>, [">= 0"])
```
Comment (DevinCodes) [inline algoliasearch-rails.gemspec:84]: Why do we remove this here? I see we still use the `kaminari` gem in `Gemfile`.

File: lib/algoliasearch-rails.rb [hunk 2]
```diff
@@ -821,7 +806,7 @@ def algolia_ensure_init(options = nil, settings = nil, index_settings = nil)
                    index_settings.delete(:slaves) ||
                    index_settings.delete('slaves')
         index_settings[used_slaves ? :slaves : :replicas] = replicas unless replicas.nil? || options[:inherit]
-        @algolia_indexes[settings].set_settings(index_settings)
+        @algolia_indexes[settings].set_settings!(index_settings)
```
Comment (DevinCodes) [inline lib/algoliasearch-rails.rb:809]: Why do we change it to `set_settings!` here instead of using the method without `!`?

**Review by DevinCodes — CHANGES_REQUESTED** [review]
I have 2 questions, other than that it looks good to me. I would wait for @julienbourdeau 's review though, since I have practically no experience with RoR 🙂

File: lib/algoliasearch-rails.rb (hunk 2 above)
Comment (chloelbn) [reply lib/algoliasearch-rails.rb:809]: I don't understand why we wouldn't wait for the settings to be updated in the first place, this is why

File: lib/algoliasearch-rails.rb [hunk 3]
```diff
@@ -821,7 +806,7 @@ def algolia_ensure_init(options = nil, settings = nil, index_settings = nil)
                    index_settings.delete(:slaves) ||
                    index_settings.delete('slaves')
         index_settings[used_slaves ? :slaves : :replicas] = replicas unless replicas.nil? || options[:inherit]
```
Comment (DevinCodes) [inline lib/algoliasearch-rails.rb:808]: These lines and the lines above still refer to the old `:slaves` setting, can you remove these in favor of `replicas`, please? 🙂

File: lib/algoliasearch-rails.rb (hunk 2 above)
Comment (DevinCodes) [reply lib/algoliasearch-rails.rb:809]: Seeing how it's used I think we can indeed wait for the settings to pass 👍

**Review by DevinCodes — APPROVED** [review]
This looks good to me 🙂  Let's see what @julienbourdeau has to say about it before merging 😄

**Review by TomKlotzPro — APPROVED** [review]
Look good to me! Really good job @chloelbn ! 
Thanks for this new major 🌹

**Pull request merged**